# Decoding the genomic regulatory syntax driving notochord development

**Author:** Clarence Mah <br>
**University:** University of California San Diego <br>
**Department:** [Bioinformatics & Systems Biology](https://bioinformatics.ucsd.edu/)

## Committee Members

* Professor Hannah Carter, *Chair*
* Professor Gene Yeo, *Co-Chair*
* Professor Nate Lewis
* Professor Samara Reck-Peterson
* Professor Bing Ren

## Abstract

Emerging genomic technologies that measure spatial information about RNA molecules promise to shed light on cell biology and function. However, most analytical techniques have primarily concentrated on spatial relationships at the multicellular and cellular scale without fully tapping into single-molecule spatial information. To address this gap, I introduce Bento, a toolkit designed for discerning spatial relationships at the subcellular scale. Bento incorporates a suite of statistical and machine learning methods within an intuitive Python programming interface, emphasizing the FAIR data management principles. To showcase its capabilities, I utilized Bento to study RNA localization changes in doxorubicin-treated cardiomyocytes profiled with spatial transcriptomics. Our findings reveal that doxorubicin-induced stress leads to the depletion of disease-associated genes in the endoplasmic reticulum, along with expression changes previously associated with doxorubicin-induced cardiotoxicity. This places the endoplasmic reticulum as a pivotal subcellular structure in the response to doxorubicin treatment. In essence, Bento emerges as a potent toolkit for the subcellular analysis of spatial transcriptomics data, paving the way for the discovery of new spatial relationships between subcellular structures and molecules. Furthermore, I have created a framework tailored to streamline image processing for spatial transcriptomics data called spotfish. Similar to Bento's ethos, spotfish is built in alignment with FAIR principles and leverages open-source standards like the Nextflow workflow language and Open Microscopy Environment file formats. Collectively, Bento and spotfish empower researchers to harness spatial transcriptomics technologies, enabling more comprehensive exploration of the spatial and molecular organization of cells at an unprecedented throughput.

## Acknowledgements 

My dissertation document was derived from templates from the following repos:
- https://github.com/mragsac/phd-dissertation
- https://github.com/stevecheckoway/ucsddissertation
